vowpalwabbit/marginal.cc

```diff
@@ -56,6 +110,29 @@ void predict_or_learn(data& sm, LEARNER::base_learner& base, example& ec)
     base.learn(ec);
   else
     base.predict(ec);
+  
+  float feature_pred = ec.pred.scalar;
+
+  if(sm.compete) { //add in the feature-based expert and normalize,
+		   //assign to example
+    //net_feature_weight = 0.; useful in case we want to try pure marginals
+    if(net_weight + net_feature_weight > 0.) 
+      average_pred += net_feature_weight * feature_pred;      
+    else {
+      net_feature_weight = 1.;
+      average_pred = feature_pred;
+    }
+    average_pred /= (net_weight + net_feature_weight);
```

Comment:
Instead of dividing, can you precompute 1. / (net_weight + net_feature_weight) then multiply?  multiplication is typically much faster than division.